terraform/main.tf

```diff
@@ -1,15 +1,44 @@
-# Configure AWS
+# Default AWS provider (ProvisionAccount for the Users account)
 provider "aws" {
-  region = "us-east-1"
+  region  = "us-east-1"
+  profile = "cool-users-provisionaccount"
+}
+
+# ProvisionEC2AMICreateRoles AWS provider for the Images account
+provider "aws" {
+  region  = "us-east-1"
+  profile = "cool-images-provisionec2amicreateroles"
+  alias   = "images-ProvisionEC2AMICreateRoles"
+}
+
+# ProvisionParameterStoreReadRoles AWS provider for the Images account
+provider "aws" {
+  region  = "us-east-1"
+  profile = "cool-images-provisionparameterstorereadroles"
+  alias   = "images-ProvisionParameterStoreReadRoles"
+}
+
+# Use aws_caller_identity with the Images account provider so we can pass the
+# Images account ID into the module below
+data "aws_caller_identity" "images" {
+  provider = aws.images-ProvisionParameterStoreReadRoles
 }
 
 module "iam_user" {
-  source = "github.com/cisagov/molecule-packer-travisci-iam-user-tf-module"
+  source = "github.com/cisagov/molecule-packer-ci-iam-user-tf-module"
+
+  providers = {
+    aws                                         = aws
+    aws.images-ProvisionEC2AMICreateRoles       = aws.images-ProvisionEC2AMICreateRoles
+    aws.images-ProvisionParameterStoreReadRoles = aws.images-ProvisionParameterStoreReadRoles
+  }
 
-  ssm_parameters = ["/vnc/username", "/vnc/password"]
-  user_name      = "test-ansible-role-vnc-server"
+  add_packer_permissions = true
```

Comment:
We don't need these permissions, but this is the only way to get the proper tagging so we can assume the role in our GitHub Actions workflow.  I don't like this.

The [cisagov/molecule-packer-ci-iam-user-tf-module](https://github.com/cisagov/molecule-packer-ci-iam-user-tf-module) needs to be split up.  It serves Packer well, but it longer serves Molecule well at all.

Reply:
I agree, but I'm thinking that we should remove the SSM parameters from this role and replace them with default values, similar to what I did in https://github.com/cisagov/ansible-role-guacamole/commit/b2a04688df700e394c957087d847731fa7dd39e1.  Then we can pull the SSM parameters when we actually build the AMI, like in https://github.com/cisagov/guacamole-packer/commit/513d9f4f8156c67748d4d7239063bf935ff3b947.

Reply:
Yes, that is much better! That way the Ansible role is more general. We should in general prefer looking up SSM parameters in Packer or cloud-init instead of Ansible.

Reply:
Advice taken in commit 7ff65f7.